zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/LazyHttpClientImpl.java

```diff
@@ -71,30 +70,30 @@
   }
 
   Endpoint getEndpoint() {
-    EndpointGroup endpointGroup = initialEndpoints.get();
-    if (endpointGroup instanceof Endpoint) return (Endpoint) endpointGroup;
+    EndpointGroup rawEndpointGroup = initialEndpoints.get();
+    // Only health-check when there are alternative endpoints. There aren't when instanceof Endpoint
+    if (rawEndpointGroup instanceof Endpoint) return (Endpoint) rawEndpointGroup;
 
-    // https://github.com/line/armeria/issues/2071 Composite endpoint groups don't extend dynamic
-    // once this is done, we can delete special-cased code in InitialEndpointSupplier about DNS, as
-    // when endpointGroup.isStatic() is false, we'd here.
-    if (endpointGroup instanceof DynamicEndpointGroup) {
-      try {
-        // Since we aren't holding up server startup, or sitting on the event loop, it is ok to
-        // block. The alternative is round-robin, which could be unlucky and hit a bad node first.
-        //
-        // We are blocking up to the connection timeout which should be enough time for any DNS
-        // resolution that hasn't happened yet to finish.
-        endpointGroup.awaitInitialEndpoints(timeoutMillis, TimeUnit.MILLISECONDS);
-      } catch (Exception e) {
-        // We'll try again next time around.
-        throw new IllegalStateException("couldn't connect any of " + endpointGroup.endpoints(), e);
-      }
-    }
+    EndpointGroup endpointGroup = rawEndpointGroup;
+    if (healthCheck.isEnabled()) endpointGroup = decorateHealthCheck(rawEndpointGroup);
 
-    if (healthCheck.isEnabled()) endpointGroup = decorateHealthCheck(endpointGroup);
+    try {
+      // Since we aren't holding up server startup, or sitting on the event loop, it is ok to
+      // block. The alternative is round-robin, which could be unlucky and hit a bad node first.
+      //
+      // We are blocking up to the connection timeout which should be enough time for any DNS
+      // resolution that hasn't happened yet to finish.
+      //
+      // TODO: When using health checks, if all hosts fail the health check, this will throw TOE
+      // See https://github.com/line/armeria/issues/2075
+      endpointGroup.awaitInitialEndpoints(timeoutMillis, TimeUnit.MILLISECONDS);
+    } catch (Exception e) {
+      // We'll try again next time around.
```

Comment:
`endpointGroup.close()` - no-op if it's not a health checked group.

Reply:
thanks